timeseries/src/autogluon/timeseries/splitter.py

```diff
@@ -0,0 +1,192 @@
+import logging
+from typing import Tuple
+
+import pandas as pd
+
+from .dataset import TimeSeriesDataFrame
+
+logger = logging.getLogger(__name__)
+
+
+class AbstractTimeSeriesSplitter:
+    """A class that handles train / validation splitting of timeseries datasets."""
+
+    def split(
+        self, ts_dataframe: TimeSeriesDataFrame, prediction_length: int
+    ) -> Tuple[TimeSeriesDataFrame, TimeSeriesDataFrame]:
+        """Split each series in the input dataset into one train and potentially multiple validation series.
+
+        The `item_id` of each validation series is a string of format `f"{idx}_[{start}:{end}]". Here `idx` is
+        the `item_id` of the input series, and `start` and `end` correspond to the slice parameters used to generate
+        the validation series.
+
+        For example, the series `val_data.loc["0_[None:-10]"]` in the validation dataset can equivalently be
+        obtained as `ts_dataframe.loc[0][None:-10]`.
+
```

Comment:
One problem with this approach is that we cannot determine if the original `item_id` was an `int` or a `str` anymore after obtaining the new ID as `f"{idx}_[{start}:{end}]`. 

This may be slightly problematic if, for example, we at some point want to retrieve the respective local model trained on `train_data` and score it on a sequence from `tuning_data` with the new index.

The only alternatives I could think of are
1. Converting the `item_id` index of `train_data` to `str` as well
2. Adding an additional index level that denotes the split (in addition to the `item_id` and `timestamp` levels)

Both of these, however, look really dangerous and can have side effects.

Reply:
I guess this shouldn't be a problem anymore with the new implementation for local models that we discussed.